Application/OpenCore/OpenCore.c

```diff
@@ -135,6 +136,7 @@ OcMain (
 
   DEBUG ((DEBUG_INFO, "OC: OcLoadNvramSupport...\n"));
   OcLoadNvramSupport (Storage, &mOpenCoreConfiguration);
+  OcReportVersion (&mOpenCoreConfiguration);
```

Comment:
Why was it moved here?

Reply:
I personally strongly dislike the idea of having config code anywhere outside of OcMainLib. Can we avoid this?

Reply:
So just to be sure I understand, you do mean you're against the idea of moving what I've moved (NVRAM config code), into a separate NVRAM lib?

In that case (pending being back at Mac/hack and actually trying it) I think we could move _some_ of this back into OcMainLib.

But the essential issue is that the new emulated NVRAM protocol now has to do some of what the normal OC NVRAM config does (some of the same code, for the same reasons). So _that_ code cannot be in OcMainLib, without having to then link OcMainLib into OpenDuet - which creates a horrible mess (with TONS of otherwise non-needed stuff having to be linked into OpenDuet (only for most of it to be discarded)), as far as I can see.

Reply:
@vit9696 - Okay I have tried to address this as a new commit on the PR (i.e. trying to make it more or less convenient so that the new commit itself will show the changes just made, while the whole PR will still show the changes against the current OC code - I guess that's obvious!).

The main change is:
 - I have ensured that no OC global config is accessed outside of OcMainLib now ... except LegacySchema/OC_NVRAM_LEGACY_MAP
 - I still can't see how to avoid that. (Although I guess we could define that structure not in the config, but then include it in the config - but I think that wouldn't be a great improvement, if any.)

Also other minor fixes of other issues mentioned below, for which I'll comment or close the relevant conversations.